lib/ansible/parsing/yaml/constructor.py

```diff
@@ -78,6 +86,18 @@ def construct_mapping(self, node, deep=False):
 
         return mapping
 
+    def construct_ansible_int(self, node, unsafe=False):
+        value = self.construct_scalar(node)
+        if value.startswith('0') and len(value) > 1:
+            if value.startswith('0x'):
+                # TODO: use 'hex object'?
+                ret = self.construct_yaml_hex(node, unsafe)
+            else:
+                ret = self.construct_yaml_octal(node, unsafe)
+        else:
+            ret = self.construct_yaml_int(node)
```

Comment:
You need to handle unsafe in the else branch.

Reply:
There shouldn't be any difference between octal, hex, and decimal for this case....

Reply:
the objects need to be different, since they hardcode the representation to use oct() or hex() and the input to int(x, 8) or int(x, 16)

Reply:
Which might mean that we should remove unsafe from all of them?

Reply:
Talking about unsafe.  Not the stuff needed for representing octal values.